saleor/graphql/warehouse/dataloaders.py

```diff
@@ -178,4 +178,6 @@ class WarehouseByIdLoader(DataLoader):
 
     def batch_load(self, keys):
         warehouses = Warehouse.objects.in_bulk(keys)
-        return [warehouses.get(UUID(warehouse_uuid)) for warehouse_uuid in keys]
+        return [
+            warehouses.get(UUID(str(warehouse_uuid))) for warehouse_uuid in keys
+        ]  # psycopg converts these UUID's once, possible bug?
```

Comment:
What should be done with this comment?

Reply:
I have removed this comment, however something sketchy is going on here. Probably there is a `UUID` type discrepancy in `load` and `batch_load`. `warehouse_uuid` is a UUID object itself,  but something like:
``` python
```

Reply:
Probably, I have found the cause.
In `order/types.py` we have:
``` python
    @staticmethod
    def resolve_warehouse(root: models.OrderEvent, info):
        if warehouse_pk := root.parameters.get("warehouse"):
            return WarehouseByIdLoader(info.context).load(warehouse_pk)
        return None
```
`parameters` is `JSONField`, thus, `warehouse_pk` will be `str`, on the other side when we pass warehouse_id (which is `UUIDField` and it is automatically casted to `UUID`). I can leave it as it is or good idea will be:
``` python
    @staticmethod
    def resolve_warehouse(root: models.OrderEvent, info):
        if warehouse_pk := root.parameters.get("warehouse"):
            return WarehouseByIdLoader(info.context).load(UUID(warehouse_pk))
        return None
```
and then we can write:
`return [warehouses.get(warehouse_uuid) for warehouse_uuid in keys]` where `keys: Iterable[UUID]`

What do you think?

Reply:
Done.